### PR TITLE
feat: History page(Add dropdown function)

### LINKED ISSUE #221: How about to use a flexible amount of scans that can be displayed on the page?
How about to use a flexible amount of scans that can be displayed on the page?

For example:
image

_Originally posted by @o-kopysov in https://github.com/Samsung/LPVS/pull/220#discussion_r1336098813_

### PR DESCRIPTION
**Pull request by quswjdgma83**
# Description

"This Pull Request is linked to Issue #221. I have developed a feature that allows users to adjust the quantity of scanned data using a dropdown method.
Fixes #221

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code cleanup/refactoring
- [ ] Documentation update
- [x] This change requires a documentation update
- [ ] CI system update
- [ ] Test Coverage update

# How Has This Been Tested?

1. Update environment variables of docker-compose.yml, src/main/resources/application.properties, frontend/.env file.

2. Build the lpvs container : docker compose build

3. Run containers: docker compose up

4. See the lpvs home page with any browser: http://localhost:7896

image

**Test Configuration**:
* Java: v11
* LPVS Release: v1.1.0

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by tiokim — COMMENTED** [review]
What about removing the `box` that wraps around `My Option`?

File: frontend/src/pages/History.jsx [hunk 1]
```diff
@@ -90,78 +93,143 @@ export const History= () => {
...
       return true;
     }
     else {
       return false;
     }
   }
 
-   const status_check=(a)=> {
-    if(a) {
+  const status_check = (a) => {
```
[the remark this answers is not in the document]
Comment (kyudori) [reply frontend/src/pages/History.jsx]: Done.

File: frontend/src/pages/History.jsx [hunk 2]
```diff
@@ -90,78 +93,143 @@ export const History= () => {
...
+      return
     }
-   }
-   console.log(page);
-   console.log(currentPage);
-   const check_page_minus =(page)=> {
-    if(page <=4) {
+  }
+
+  const check_page_minus = (page) => {
```
[the remark this answers is not in the document]
Comment (kyudori) [reply frontend/src/pages/History.jsx]: Done.

File: frontend/src/pages/History.jsx [hunk 3]
```diff
@@ -90,78 +93,143 @@ export const History= () => {
...
+  const [currentPage, setCurrentPage] = useState(1);
 
-   console.log(pageCount);
+  useEffect(() => {
+    setPageCount(Math.ceil(lpvsHistories?.count / selectedSize));
+  }, [lpvsHistories?.count, selectedSize]);
 
-   const check_page_plus =()=> {
-    if(currentPage <= pageCount) {
+  const check_page_plus = () => {
```
[the remark this answers is not in the document]
Comment (kyudori) [reply frontend/src/pages/History.jsx]: Done.

File: frontend/src/pages/History.jsx [hunk 4]
```diff
@@ -25,6 +25,12 @@ export const History= () => {
   const [isLoggedIn, setIsLoggedIn] = useState(false);
   const [username, setUsername] = useState("");
 
+  const [selectedSize, setSelectedSize] = useState(null);
+
+  useEffect(() => {
+    setSelectedSize(5);
+  }, []);
+
```
[the remark this answers is not in the document]
Comment (kyudori) [reply frontend/src/pages/History.jsx]: Done.

**Comment by kyudori** [comment]
> What about removing the `box` that wraps around `My Option`?
> 

Remove MyOptionBox and Move DropdownBox. 
image

File: frontend/src/css/History_style.css [hunk 5]
```diff
@@ -5,34 +5,26 @@
  * found in the LICENSE file.
  */
 
-.history-send {
+ .history-send {
```
Comment (tiokim) [inline frontend/src/css/History_style.css:8]: Please remove the empty space.

**Review by tiokim — APPROVED** [review]
LGTM! Thank you for your diligent contributions on LPVS!

File: frontend/src/css/History_style.css (hunk 5 above)
Comment (kyudori) [reply frontend/src/css/History_style.css]: Done. Thank you!

**Review by o-kopysov — APPROVED** [review]
+1

**Pull request merged**